### PR TITLE
Repro #36122 - Table Metadata having 2 foreign keys from the same table pointing to the same key cause implicit joins to only show 1

### LINKED ISSUE #36122: Table Metadata having 2 foreign keys from the same table pointing to the same key cause implicit joins to only show 1
### Describe the bug

When you have a table in which you got two columns setup as foreign key types and pointing to the same key cause the GUI editor to only show 1 of them as an implicit join and ignore completely the other

### To Reproduce

1. Go to Admin -> Table Metadata -> Sample DB -> Reviews -> Set the Rating as Foreign Key to Product ID (so it matched the Product ID one)

image

2. Go to New Question -> Reviews -> Click on Group By -> Notice only the Product is Visible and there is no Rating option

image

### Expected behavior

Both implicit joins are visible and selectable 

### Logs

None that are relevant

### Information about your Metabase installation

```JSON
47.8 and `master`
```

### Severity

It's not a blocking issue since a user can still manually perform the join from the join option of the GUI but it's annoying

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #36122

### CODE REVIEW

File: e2e/test/scenarios/admin/datamodel/metadata.cy.spec.js [hunk 1]
```diff
@@ -132,6 +133,24 @@ describe("scenarios > admin > datamodel > metadata", () => {
...
+
+    openReviewsTable({ mode: "notebook" });
+    summarize({ mode: "notebook" });
+    getNotebookStep("summarize")
+      .findByText("Pick a column to group by")
+      .click();
+
+    popover().within(() => {
+      cy.findAllByText("Rating").should("have.length", 2);
+      cy.get(".List-section-header").should("have.length", 3);
```
Comment (uladzimirdev) [inline e2e/test/scenarios/admin/datamodel/metadata.cy.spec.js:150]: why so and not `findByText` with explicit values?

Comment (kamilmielnik) [reply e2e/test/scenarios/admin/datamodel/metadata.cy.spec.js:152]: Added more assertions: 16a82e26534a56ce456327e62a9f18b3c0b6130a

Comment (uladzimirdev) [reply e2e/test/scenarios/admin/datamodel/metadata.cy.spec.js:152]: I meant a more general like `cy.findByText("Review").should('exist')`, but this should work as well

**Review by uladzimirdev — APPROVED** [review]

**Comment by uladzimirdev** [comment]
can we run a stress test for it?

File: e2e/test/scenarios/admin/datamodel/metadata.cy.spec.js (hunk 1 above)
Comment (kamilmielnik) [reply e2e/test/scenarios/admin/datamodel/metadata.cy.spec.js:152]: We're testing for "Rating" which should appear both as section header and as an item within one of the sections

**Comment by kamilmielnik** [comment]
> can we run a stress test for it?

All green: https://github.com/metabase/metabase/actions/runs/7653758434

**Pull request merged**